src/components/ComboBox/sgds-combo-box.ts

```diff
@@ -0,0 +1,120 @@
+import { ScopedElementsMixin } from "@open-wc/scoped-elements";
+import { html } from "lit";
+import { property, state } from "lit/decorators.js";
+import { ref } from "lit/directives/ref.js";
+import { DropdownElement } from "../../base/dropdown-element";
+import { defaultValue } from "../../utils/defaultvalue";
+import { SgdsDropdownItem } from "../Dropdown";
+import { SgdsInput } from "../Input";
+import styles from "./sgds-combo-box.scss";
+
+/**
+ * @summary ComboBox component is used for users to make one or more selections from a list.
+ *
+ * @event sgds-change - Emitted when the combo box's selected value changes.
+ */
+export class SgdsComboBox extends ScopedElementsMixin(DropdownElement) {
+  static styles = [DropdownElement.styles, styles];
+  static get scopedElements() {
+    return {
+      "sgds-input": SgdsInput,
+      "sgds-dropdown-item": SgdsDropdownItem
+    };
+  }
+  constructor() {
+    super();
+    /**@internal */
+    this.modifierOpt = [
+      {
+        name: "offset",
+        options: {
+          offset: [0, 10]
+        }
+      }
+    ];
+  }
+  /** The input's label  */
+  @property({ reflect: true }) label = "";
+  /** The input's hint text below the label */
+  @property({ reflect: true }) hintText = "";
+  /**The input's name attribute */
+  @property({ reflect: true }) name: string;
+  /**The input's placeholder text. */
+  @property({ type: String, reflect: true }) placeholder = "placeholder";
+  /**Autofocus the input */
+  @property({ type: Boolean, reflect: true }) autofocus = false;
+  /**Disables the input. */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+  /**Makes the input a required field. */
+  @property({ type: Boolean, reflect: true }) required = false;
+  /**Makes the input readonly. */
+  @property({ type: Boolean, reflect: true }) readonly = false;
+  /**The input's value attribute. */
+  @property({ reflect: true }) value = "";
+  /**Gets or sets the default value used to reset this element. The initial value corresponds to the one originally specified in the HTML that created this element. */
+  @defaultValue()
+  defaultValue = "";
+  @property({ type: Array }) menuList: string[] = [];
+
+  private _getFilteredMenuList(inputValue: string) {
+    return this.menuList.filter(n => {
+      const nLowerCase = n.toLowerCase();
+      const valueLower = inputValue.toLowerCase();
+      return nLowerCase.startsWith(valueLower);
+    });
+  }
+
+  private _handleInputChange(e: CustomEvent) {
+    this.value = (e.target as SgdsInput).value;
```

Comment:
sgds-input is already emitted from `sgds-input` component within `sgds-combo-box`. Do I still need to emit it here? If not sgds-input will be emitted twice

Reply:
Also, this same issue comes up for sgds-change. I know I emitted `sgds-change` event in `_handleSelectChange` which is called when a dropdown item is selected. But I realised that `sgds-select` event is already emitted from the parent (`DropdownElement`). So I actually don't need to emit `sgds-change` in ComboBox.

Reply:
oh yes cos on SgdsElement the event bubbling and composed is set to true. Yes so actually dont need to emit sgds-change a second time. but i still need you to add the jsdocs for both events inside combobox so that custom element analzyer can parsed them for storybook and react component

Reply:
alright